### PR TITLE
[Actions/Sources] Zoho Campaign

### LINKED ISSUE #7408: [Components] Zoho Campaigns
### App
Zoho Campaigns
https://www.zoho.com/campaigns/

Triggers:
1. New campaign
2. New contact

Actions:
1. Add contact to mailing list
https://www.zoho.com/campaigns/help/developers/contact-subscribe.html
2. Create campaign: Create a new campaign
https://www.zoho.com/campaigns/help/developers/create-campaign.html

### PR DESCRIPTION
**Pull request by vellames**
Closes #7408 
### <samp>🤖 Generated by Copilot at 0c44114</samp>

This pull request adds a new Zoho Campaigns app to the Pipedream components, with two action components and two source components. The action components allow users to create campaigns and add contacts to mailing lists, while the source components emit events when new campaigns and contacts are created. The pull request also adds the app module, the common module, and the package.json file for the Zoho Campaigns app, and updates the pnpm-lock.yaml file with a new dependency.

### <samp>🤖 Generated by Copilot at 0c44114</samp>

> _`Zoho Campaigns` app_
> _New components and methods_
> _Autumn harvest time_

### <samp>🤖 Generated by Copilot at 0c44114</samp>

*  Add two new action components to the Zoho Campaigns app: `add-contact-to-mailing-list` and `create-campaign` ([link](https://github.com/PipedreamHQ/pipedream/pull/7540/files?diff=unified&w=0#diff-035a82f1b8b6dfe0d4328c2ee08c65aa75ce4190af62e62211512df15edde435R1-R93), [link](https://github.com/PipedreamHQ/pipedream/pull/7540/files?diff=unified&w=0#diff-5493cc5e9871d0d702330df64231bb55b9057d33b015a0a9b802e93ff12e2971R1-R73))
*  Add two new source components to the Zoho Campaigns app: `new-campaign` and `new-contact` ([link](https://github.com/PipedreamHQ/pipedream/pull/7540/files?diff=unified&w=0#diff-38ae4674df32ce355d7becd2d940a051fde1718fce55e80dfcdc3978f3f63a97R1-R40), [link](https://github.com/PipedreamHQ/pipedream/pull/7540/files?diff=unified&w=0#diff-5884a80f470f08736d3d1ad5352bbd9937dd1c242a4d9c7b49fe0fa0dfbae26fR1-R49))
*  Add a common module to define the shared props and methods for the source components ([link](https://github.com/PipedreamHQ/pipedream/pull/7540/files?diff=unified&w=0#diff-db016c65cc8a9397ef7ae30f7d4d227b96e132e06349d288690d76c483902c8aR1-R23))
*  Add an app module to define the app type, auth, prop definitions, and methods for the Zoho Campaigns app ([link](https://github.com/PipedreamHQ/pipedream/pull/7540/files?diff=unified&w=0#diff-d820257f9919753f89d60cf6949c798788d21721cfbedff5b248cce48d52ccfbR1-R140))
*  Update the version of the Zoho Campaigns app package from 0.0.1 to 0.1.0 ([link](https://github.com/PipedreamHQ/pipedream/pull/7540/files?diff=unified&w=0#diff-468f99ad55184781b4660a8b13a2539f417d4374849980db1e73cb4debe58f06L3-R3))
*  Add a dependency to the `components/moxie` package in the `pnpm-lock.yaml` file ([link](https://github.com/PipedreamHQ/pipedream/pull/7540/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR2936-R2937))
*  Delete the unused files `zoho_campaigns.app.ts` and `.gitignore` ([link](https://github.com/PipedreamHQ/pipedream/pull/7540/files?diff=unified&w=0#diff-8359e33b9e6f0f140d6ca464db473ca41068a4159f4f09e22940b096bc03fcbf), [link](https://github.com/PipedreamHQ/pipedream/pull/7540/files?diff=unified&w=0#diff-b49b6e219dde0c49bc5b60303184e6e0bbfc2d276c5e10e6002b15aceebddc66))

### CODE REVIEW

**Comment by alysonturing** [comment]
Some checks were not successful

File: components/zoho_campaigns/actions/add-contact-to-mailing-list/add-contact-to-mailing-list.mjs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      description: "Contact last name can be added.",
+      optional: true,
+    },
+    source: {
+      type: "string",
+      label: "Source",
+      description: "Contact source can be added.",
+      optional: true,
+    },
+    topic_id: {
```
Comment (alysonturing) [inline components/zoho_campaigns/actions/add-contact-to-mailing-list/add-contact-to-mailing-list.mjs:42]: use cammelCase

File: components/zoho_campaigns/actions/create-campaign/create-campaign.mjs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  version: "0.0.1",
+  description: "You can create a campaign using this API. Using this API, you can set the campaign name, subject line, sender address; choose the intended mailing list.. [See the documentation](https://www.zoho.com/campaigns/help/developers/create-campaign.html)",
+  props: {
+    app,
+    campaignname: {
+      type: "string",
+      label: "Campaign Name",
+      description: "The name of the campaign",
+    },
+    from_email: {
```
Comment (alysonturing) [inline components/zoho_campaigns/actions/create-campaign/create-campaign.mjs:16]: use cammelCase

**Review by alysonturing — CHANGES_REQUESTED** [review]
some props are not using cammelCase

File: components/zoho_campaigns/actions/add-contact-to-mailing-list/add-contact-to-mailing-list.mjs (hunk 1 above)
Comment (vellames) [reply components/zoho_campaigns/actions/add-contact-to-mailing-list/add-contact-to-mailing-list.mjs:42]: I am using snake case to be able to use destructor in `run`. I would need to write more code if I use cammelCase for everything

File: components/zoho_campaigns/actions/create-campaign/create-campaign.mjs (hunk 2 above)
Comment (vellames) [reply components/zoho_campaigns/actions/create-campaign/create-campaign.mjs:16]: I am using snake case to be able to use destructor in `run`. I would need to write more code if I use cammelCase for everything

File: components/zoho_campaigns/actions/create-campaign/create-campaign.mjs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      type: "string",
+      label: "From Email",
+      description: "The email address from which the campaign is sent",
+    },
+    subject: {
+      type: "string",
+      label: "Subject",
+      description: "The subject of the campaign",
+    },
+    content_url: {
```
[the remark this answers is not in the document]
Comment (vellames) [reply components/zoho_campaigns/actions/create-campaign/create-campaign.mjs:26]: I am using snake case to be able to use destructor in `run`. I would need to write more code if I use cammelCase for everything

**Review by michelle0927 — DISMISSED** [review]

**Review by alysonturing — DISMISSED** [review]

**Review by alysonturing — APPROVED** [review]
LGTM

**Pull request merged**